### ISSUE TITLE
Publishing sources and javadoc should be optional

### ISSUE THREAD

**Issue by rpalcolea** [jenkinsci/gradle-jpi-plugin#138]
Hi folks,

Looking `JpiPlugin`, configuration for sourcesJar and javadocJar are always in place: https://github.com/jenkinsci/gradle-jpi-plugin/blob/07049b160300b41d6dbe28bb83b4c8427e8d3288/src/main/groovy/org/jenkinsci/gradle/plugins/jpi/JpiPlugin.groovy#L107

I think this should be optional because this would collide with other plugins doing the same. For example an opinionated plugin for publishing, example: https://github.com/nebula-plugins/nebula-publishing-plugin

Thoughts?

**Comment by sghill** [comment]
Hi @rpalcolea, thanks for the question!

Would it be sufficient for this plugin to switch to using the new Gradle 6.0 javadoc and sources features?

```
java {
  withJavadocJar()
  withSourcesJar()
}
```

I see in [the docs](https://docs.gradle.org/6.0/javadoc/org/gradle/api/plugins/JavaPluginExtension.html#withSourcesJar--) it shouldn’t create a conflicting task if one is present.

**Comment by rpalcolea** [comment]
oh I haven't tried against the RC. I had troubles with the old approach of registering task.

With that said if two plugins do 

```
java {
  withJavadocJar()
  withSourcesJar()
}
```

Everything should be fine.

With that said, there might be opinions that created tasks for generating those sources with even another task name. So you would endup with different tasks creating the same artifacts and Gradle will fail because it finds duplicates.

I'm working on this proposal: https://github.com/gradle/gradle/pull/12216

Unfortunately, this is the current state for us: https://github.com/nebula-plugins/nebula-publishing-plugin/blob/master/src/main/groovy/nebula/plugin/publishing/publications/SourceJarPlugin.groovy#L37

with `sourceJar` and the `withSourcesJar` coming from this plugin or any other, we would end up with both. 

My proposal for Gradle folks is to allow custom naming of the sources/javadoc jar tasks so migration to core gradle feature becomes easier. I've seen usage on several projects this way: https://github.com/search?l=Gradle&q=sourceJar&type=Code

**Comment by sghill** [comment]
Fixed by #140
